### PR TITLE
Standardize literals to snake_case and add cast_embed location context + client context

### PR DESCRIPTION
**Pull request by cmlad**

### CODE REVIEW

File: packages/frame-core/src/types.ts [hunk 1]
```diff
@@ -18,7 +18,15 @@ export type AccountLocation = {
   description: string;
 };
 
-export type FrameLocationNotificationContext = {
+export type FrameLocationContextCastEmbed = {
+  type: "cast_embed";
+  cast: {
+    fid: number;
+    hash: string;
```
Comment (cmlad) [inline packages/frame-core/src/types.ts:25]: starting conservative, no `text` due to then also having to have mentions, mentionsPositions, etc

File: packages/frame-core/src/types.ts [hunk 2]
```diff
@@ -40,16 +58,12 @@ export type FrameContext = {
     pfpUrl?: string;
   };
   location?: FrameLocationContext;
```
Comment (deodad) [inline packages/frame-core/src/types.ts:60]: slight preference for explicit global context vs undefined

Comment (cmlad) [reply packages/frame-core/src/types.ts:65]: could you check now, called it `launcher` instead of home so it's more generic

Comment (deodad) [reply packages/frame-core/src/types.ts:65]: good call

File: packages/frame-core/src/types.ts [hunk 3]
```diff
@@ -40,38 +63,36 @@ export type FrameContext = {
...
+  client: {
```
Comment (deodad) [inline packages/frame-core/src/types.ts:66]: I think this works.

consider optional `clientFid` so clients can self ID
`frameAdded` can be simplified to `added`
`notificationDetails` could be `notificationToken`

Comment (cmlad) [reply packages/frame-core/src/types.ts:66]: we need the url there as well, `notificationDetails` is what we use in webhooks and the `addFrame` result so prefer to keep it for consistency

**Pull request merged**